### PR TITLE
Change nGraph serializer behavior for dynamic dimensions

### PR DESCRIPTION
**Pull request by achetver**
### Details:
 - Change `dynamic_to_static` to `dinamic_to_undefined` function to use -1 values instead of upperbounds in serialized IRs.

### Tickets:
 - 68057

### CODE REVIEW

**Comment by ilyachur** [comment]
Can we run additional checks in order to avoid regressions after this change?

**Comment by achetver** [comment]
> Can we run additional checks in order to avoid regressions after this change?

Yes, of course. I'm trying to use custom e2e to reach results. If you have any ideas about how we should check this change, share them, please.

**Comment by achetver** [comment]
There are no new failed tests as I can see from comparison of custom nightly e2e dynamism tests run and one of latest nightly tests reports.

File: src/core/src/pass/serialize.cpp [hunk 1]
```diff
@@ -724,9 +724,9 @@ bool resolve_dynamic_shapes(const ngraph::Function& f) {
         op->validate_and_infer_types();
         clone_op->validate_and_infer_types();
 
-        // dynamic_to_static function converts dynamic dimensions to static using
-        // upperbound (get_max_length) dimension value.
-        auto dynamic_to_static = [&op](const PartialShape& shape) -> PartialShape {
+        // dynamic_to_static function converts dynamic dimensions to undefined using
```
Comment (rkazants) [inline src/core/src/pass/serialize.cpp:727]: nit: `dynamic_to_static function` -> `dynamic_to_undefined function`

**Comment by rkazants** [comment]
> There are no new failed tests as I can see from comparison of custom nightly e2e dynamism tests run and one of latest nightly tests reports.

Let me run this change on DLB and please check with POT as well

File: src/core/src/pass/serialize.cpp [hunk 2]
```diff
@@ -724,9 +724,9 @@ bool resolve_dynamic_shapes(const ngraph::Function& f) {
...
+        // -1 dimension value.
```
Comment (rkazants) [inline src/core/src/pass/serialize.cpp:728]: also, I think we need to have some test for this rountine

File: src/core/src/pass/serialize.cpp [hunk 3]
```diff
@@ -696,74 +696,6 @@ bool has_dynamic_output(const std::shared_ptr<Node>& n) {
     return false;
 }
 
-bool resolve_dynamic_shapes(const ngraph::Function& f) {
```
Comment (GlebKazantaev) [inline src/core/src/pass/serialize.cpp:699]: Please add test for Serialization with dynamic dims.

Comment (achetver) [reply src/core/src/pass/serialize.cpp:699]: Where is the place to add it?

Comment (popovaan) [reply src/core/src/pass/serialize.cpp:699]: openvino/inference-engine/tests/functional/inference_engine/ir_serialization/
Look at partial_shape_serialization.cpp, partial_shape_deserialization.cpp as examples.

**Comment by rkazants** [comment]
@KodiaqQ, Is the POT tool ready for IRs with `-1` to mark dynamic dimensions? How can this change affect the POT tool?

**Comment by AlexKoff88** [comment]
> @KodiaqQ, @AlexKoff88, could you please tell if the POT tool is ready for IRs with `-1` to mark dynamic dimensions? How can this change affect the POT tool?

We got a request to validate this. It seems that it should not impact internal POT functionality but there are 3rd party components, like IRReader, AC, etc., that are used. So only validation will show the overall impact.

File: src/core/src/pass/serialize.cpp (hunk 3 above)
Comment (achetver) [reply src/core/src/pass/serialize.cpp:715]: As far as I can understand this functionality covered by tests from here: `src/tests/functional/inference_engine/ir_serialization/partial_shape_serialization.cpp`. And looks like we already test dynamic dims and these tests passed without changes.

**Review by rkazants — APPROVED** [review]
there is no regression in DLB

**Pull request merged**